server/circleci.go

```diff
@@ -74,22 +76,117 @@ func (s *Server) triggerCircleCiIfNeeded(pr *model.PullRequest) {
 	mlog.Info("Triggered circleci", mlog.String("repo", pr.RepoName), mlog.Int("pr", pr.Number), mlog.String("fullname", pr.FullName))
 }
 
-func (s *Server) triggerEnterprisePipeline(pr *model.PullRequest, eeBranch string, triggerBranch string) error {
+type PipelineTriggeredResponse struct {
+	Number    int       `json:"number"`
+	State     string    `json:"state"`
+	Id        string    `json:"id"`
+	CreatedAt time.Time `json:"created_at"`
+}
+
+func (s *Server) triggerEnterprisePipeline(pr *model.PullRequest, eeBranch string, triggerBranch string) (string, error) {
 	body := strings.NewReader(`branch=` + eeBranch + `&parameters[external_branch]=` + triggerBranch + `&parameters[external_sha]=` + pr.Sha + `&parameters[external_pr]=` + strconv.Itoa(pr.Number))
 	req, err := http.NewRequest("POST", "https://circleci.com/api/v2/project/gh/"+s.Config.Org+"/"+s.Config.EnterpriseReponame+"/pipeline", body)
 	if err != nil {
-		return err
+		return "", err
 	}
 	req.SetBasicAuth(os.ExpandEnv(s.Config.CircleCIToken), "")
 	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
 
-	resp, err := http.DefaultClient.Do(req)
+	client := http.Client{
+		Timeout: 20 * time.Second,
+	}
+	resp, err := client.Do(req)
```

Comment:
We can set the timeout in the request itself, using `http.NewRequestWithContext` or the `req.WithContext` method. Initializing a new http.Client every time has a cost since a client is a fat object maintaining persistent TCP connections with many hosts and reuses them.